cmd/debugger/main.go

```diff
@@ -89,7 +91,15 @@ func interactiveMode(ctx context.Context, remoteConsoleAddr string, cmdBuilder f
 
 	conn, err := net.Dial("tcp", remoteConsoleAddr)
 	if err != nil {
-		return errors.Wrap(err, "failed to connect to remote debugger")
+		// If the IP we derived at start fails, try the reserved gateway address for our internal CNI network.
+		// The CIDR range for this can be found in buildkitd/cni-conf.json.template, so if that ever changes, we need to change it here, too.
+		// Relevant CNI docs for the host-local ipam module: https://www.cni.dev/plugins/current/ipam/host-local/
+		// tl;dr we can assume the gateway is at .1 in the subnet.
+		log.Debug(fmt.Sprintf("Failed to connect to \"%s\", trying internal CNI fallback \"%s\".", remoteConsoleAddr, cniFallbackHost))
+		conn, err = net.Dial("tcp", cniFallbackHost)
+		if err != nil {
+			return errors.Wrap(err, "failed to connect to remote debugger")
+		}
```

Comment:
Could we switch completely to the CNI one only? We'd be able to get rid of the port assignment on the host.

Or at the very least, try the CNI one address first, and if that doesn't work, try the old one.

Reply:
We could. I was just trying to preserve the old behavior first. Ill do that now.

Reply:
Done.